butterfly-cli/src/test/resources/log4j2-test.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Configuration for test purpose -->
+<configuration name="Test" packages="com.paypal.butterfly.cli.logging">
+	<Appenders>
+		<Console name="CONSOLE" target="SYSTEM_OUT" follow="true">
+			<PatternLayout pattern="%msg%n" />
+		</Console>
+
+		<RandomAccessFile name="FILE" fileName="${LogFileDefiner:logfile}" >
+			<PatternLayout>
+				<pattern>[%d{HH:mm:ss.SSS}] [%-4level] %msg%n</pattern>
+			</PatternLayout>
+		</RandomAccessFile>
+	</Appenders>
+
+	<Loggers>
+		<Logger name="com.paypal.butterfly.cli" level="INFO">
+			<AppenderRef ref="CONSOLE" />
+		</Logger>
+		<Root level="ERROR">
+			<AppenderRef ref="FILE" />
+		</Root>
+	</Loggers>
+</configuration>
```

Comment:
I need some guidance from you here. This configuration file is used for test purpose. The difference between this and the [production configuration](https://github.com/paypal/butterfly/pull/367/files#diff-3b2b25e9fb51ff13c63c97f5cfde8824e921ad41f2cb52f6bb7edc39ee681b2a) file is that loggers are synchronous here. I did this because the production loggers are async, many test cases were failing. But there is something that concerns me.

- Now there are two configuration files. One for test and one for production. There is possibility they might diverse. For example, if someone change the test file but doesn't make the same change in the production file. To guard against this I have used [this test](https://github.com/paypal/butterfly/pull/367/files#diff-1e04d91dccfdf1522d572997ab5dadfe1d8dc8c82c6c68b81e6e4fda550e4033) which uses the production configuration file. Do you think another approach would have been better? How can we test async loggers?

- Log4j2 has an option called [`XInclude`](https://logging.apache.org/log4j/2.x/manual/configuration.html#XInclude) which allows to add fragments of configuration from different file. This can improve the situation. For example, I can keep the common configuration for production and test in one file and then include the configuration that is varying, separately. The production config is in `butterfly-cli/src/main/resources` folder. Whereas test configuration is in `butterfly/cli/src/test/resources` folder. I am not able to use `XInclude` to include configuration from production folder because relative referencing is not working in `href`. Do you know how to do it?

Reply:
I see your point, thanks for sharing it. Indeed there is a risk we introduce inconsistencies as we maintain these log files. However, after this PR is merged, unless if we find any bug with this solution, there will be very rare moments when we have to go back here. Also, trying to solve this problem might be too complicated, and not worth it compared to the value it would provide. Having said that, I think it is ok to have this as it is now. So, I think we will be fine.

Still, just to help out a little. Please add a comment to the top of both files explaining the difference between them, why we need a test one, and reminding code maintainers to make sure they are "in sync" whenever any change is necessary.

Thank you very much.

Reply:
Great. Will do. Thank you. :-)